GNUmakefile

```diff
@@ -415,7 +424,17 @@ clean:
 	$(MAKE) -C qemu_mode/unsigaction clean
 	$(MAKE) -C qemu_mode/libcompcov clean
 	$(MAKE) -C src/third_party/libradamsa/ clean
-	-rm -rf unicorn_mode/unicornafl
+	rm -rf qemu_mode/qemu-3.1.1
+ifeq "$(IN_REPO)" "1"
+	$(MAKE) -C unicorn_mode/unicornafl clean || true
+else
+	rm -rf qemu_mode/qemu-3.1.1.tar.xz
+	rm -rf unicorn_mode/unicornafl
+endif
+
+deepclean:	clean
+	rm -rf qemu_mode/qemu-3.1.1.tar.xz
+	rm -rf unicorn_mode/unicornafl
```

Comment:
technically wrong if in git, because then unicornafl is a submodule, removing it changes the state of the repo. Resetting the git repo might be the better/more correct solution?

Reply:
deepclean can also be called by people not having git. then maybe `rm -rf unicorn_mode/unicornafl/*`?

Reply:
no, there is no unicornafl subdirectory in the repo so deepclean is correct

Reply:
For the git repo, unicornafl is a submodule - so it's there if you recursively clone or manually pull the submodules using `git submodule init && git submodule update`.
In this case, a way to "deep clean" is to do a `git submodule foreach --recursive git clean -xfd` (reset all changed files for each submodule).
If you simply remove the folder, git will think you changed the aflplusplus repo to no longer contain the submodule (leaving the parent repo in a changed state)
Another way is probably to clean the parent repo after the removal, that way the submodule should also be back (git `git reset --hard` or sthm)
Sorry for the confusion. Should I try to come up with a solution?

Reply:
ah understood. I dont use git :) yeah if you could find a solution ...

Reply:
should(tm) be solved, please feel free to try without git :)